share/availability/light/availability.go

```diff
@@ -169,6 +175,59 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 	return nil
 }
 
+// Prune deletes samples and all sampling data corresponding to provided header from store.
+// The operation will remove all data that ShareAvailable might have created
+func (la *ShareAvailability) Prune(ctx context.Context, h *header.ExtendedHeader) error {
+	dah := h.DAH
+	if share.DataHash(dah.Hash()).IsEmptyEDS() {
+		return nil
+	}
+
+	// Prevent multiple sampling sessions for the same header height
+	release, err := la.activeHeights.StartSession(ctx, h.Height())
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	key := datastoreKeyForRoot(dah)
+	la.dsLk.RLock()
+	data, err := la.ds.Get(ctx, key)
+	la.dsLk.RUnlock()
+	if errors.Is(err, datastore.ErrNotFound) {
+		// nothing to prune
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get sampling result: %w", err)
```

Comment:
cosmetic nit: in Shwap, we've been making errors like `getting sampling result` as writing `failed` everywhere redundant and bloats error message.